### ISSUE TITLE
o-forms: improve error handling for date inputs - ID: DAC_Error_Handling_01

### ISSUE THREAD

**Issue by JakeChampion** [Financial-Times/origami#517]
# Issue
This is from page 75 of the [audit report](https://drive.google.com/file/d/17wq6b0CIX41RA4muBFDzUDxE0-qzHBIH/view)

Issue can be found on: <https://www.ft.com/__origami/service/build/v3/demo?component=o-forms@9.0.1&demo=inverse-form&system_code=origami-repo-data&brand=core>

When users enter incorrect data into the date input fields, the error handling is confusing. When users begin to enter into the first ‘DD’ input field a message appears which reads ‘Please fill out this field with the required format (DD/MM/YYYY)’. For sighted users they are able to see this message, but it is not announced to screen reader users.
           
When users enter data into the second input field ‘MM’ the message "Please fill out this field with the required format (DD/MM/YYYY)" disappears and the colour of the input field changes colour to green which would indicate to sighted users that they have entered the information correctly.
But if the user enters the date in the wrong format an error summary appears at the top of the page as expected. It would benefit all users if the use of colour in the input fields was removed as it may be confusing for users who may think the green colour indicates successful validation.

 
# Proposed Solution:
Remove the JavaScript functions which add colour and the error message to the input fields. Use inline HTML error messages in conjunction with the error summary. Further information can be found in the Error messages – Date inputs, Error message and Error summary in the GOV.UK Design System.

**Comment by JakeChampion** [comment]
This issue is also present on <https://www.ft.com/__origami/service/build/v3/demo?component=o-forms@9.0.1&demo=interactive&system_code=origami-repo-data&brand=master> and found on page 150 of the audit report

ID: DAC_Error_Handling_02
